pkg/agent/controller/l7visibility/l7visibility_controller.go

```diff
@@ -0,0 +1,400 @@
+	l7c.queue.Add(podNN)
+}
+
+func (l7c *L7VisibilityController) updatePod(oldObj interface{}, obj interface{}) {
+	oldPod := oldObj.(*v1.Pod)
+	updatedPod := obj.(*v1.Pod)
+	if !isValidPod(updatedPod) {
+		return
+	}
+	oldAnnotation, _ := oldPod.Annotations[types.L7VisibilityAnnotationKey]
+	updatedAnnotation, updatedAnnotationOk := updatedPod.Annotations[types.L7VisibilityAnnotationKey]
+	if oldAnnotation == updatedAnnotation {
+		if !updatedAnnotationOk {
+			return
+		}
+		if oldPod.Status.PodIP == updatedPod.Status.PodIP {
+			return
+		}
+	}
+
+	klog.V(2).InfoS("Processing Pod UPDATE event", "Pod", klog.KObj(updatedPod))
+	podNN := k8s.NamespacedName(updatedPod.Namespace, updatedPod.Name)
+	l7c.queue.Add(podNN)
+}
+
+func (l7c *L7VisibilityController) deletePod(obj interface{}) {
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.ErrorS(nil, "Received unexpected object", "object", obj)
+			return
+		}
+		pod, ok = deletedState.Obj.(*v1.Pod)
+		if !ok {
+			klog.ErrorS(nil, "DeletedFinalStateUnknown contains non-Pod object", "object", deletedState.Obj)
+			return
+		}
+	}
+	if _, ok := pod.Annotations[types.L7VisibilityAnnotationKey]; !ok {
+		if !l7c.namespaceAnnotationExists(pod) {
+			return
+		}
+	}
+
+	klog.V(2).InfoS("Processing Pod DELETE event", "Pod", klog.KObj(pod))
+	podNN := k8s.NamespacedName(pod.Namespace, pod.Name)
+	l7c.queue.Add(podNN)
+}
+
+func (l7c *L7VisibilityController) namespaceAnnotationExists(pod *v1.Pod) bool {
+	podNamespace, err := l7c.namespaceLister.Get(pod.Namespace)
+	if err != nil {
+		return false
+	}
+	_, ok := podNamespace.Annotations[types.L7VisibilityAnnotationKey]
+	return ok
+}
+
+func isValidPod(pod *v1.Pod) bool {
+	return pod.Status.PodIP != "" && !pod.Spec.HostNetwork
+}
+
+func (l7c *L7VisibilityController) addNamespace(obj interface{}) {
+	ns := obj.(*v1.Namespace)
+	if _, ok := ns.Annotations[types.L7VisibilityAnnotationKey]; !ok {
+		return
+	}
+	klog.V(2).InfoS("Processing Namespace ADD event", "Namespace", klog.KObj(ns))
+	affectedPods := l7c.getNonAnnotatedPodsFromNS(ns)
+	for _, pod := range affectedPods {
+		podNN := k8s.NamespacedName(pod.Namespace, pod.Name)
+		l7c.queue.Add(podNN)
+	}
+}
+
+func (l7c *L7VisibilityController) updateNamespace(oldObj, obj interface{}) {
+	oldNS := oldObj.(*v1.Namespace)
+	updatedNS := obj.(*v1.Namespace)
+	oldAnnotation, _ := oldNS.GetAnnotations()[types.L7VisibilityAnnotationKey]
+	updatedAnnotation, _ := updatedNS.GetAnnotations()[types.L7VisibilityAnnotationKey]
+	if oldAnnotation == updatedAnnotation {
+		return
+	}
+
+	klog.V(2).InfoS("Processing Namespace UPDATE event", "Namespace", klog.KObj(updatedNS))
+
+	affectedPods := l7c.getNonAnnotatedPodsFromNS(updatedNS)
+	for _, pod := range affectedPods {
+		podNN := k8s.NamespacedName(pod.Namespace, pod.Name)
+		l7c.queue.Add(podNN)
+	}
+}
+
+func (l7c *L7VisibilityController) getNonAnnotatedPodsFromNS(ns *v1.Namespace) []*v1.Pod {
```

Comment:
s/getNonAnnotatedPodsFromNS/getNonAnnotatedPodsFromNamespace

for consistency with your other method names